docs/arch/adr-013-rabbitmq-sidekiq-removal.md

```diff
@@ -0,0 +1,33 @@
+# Decision record: Removing Sidekiq from RabbitMQ queue consumers
+
+**Date:** 2025-04-07
+
+This supersedes [ADR-001][].
+
+## Context
+After investigation following a recent incident where Redis was running out of memory, it was determined that the problem lay in the queue consumer code in this codebase.
+
+When consuming a job off the RabbitMQ queue, we immediately push the payload into a Sidekiq job for asynchronous processing, essentially moving the processing from one queue to another. This poses a particular problem when bulk reindexing of documents takes place, as thousands of jobs are pushed to Sidekiq (and therefore Redis).
+
+When the queue consumer code was written, there were no concurrency features available in our RabbitMQ consumer gem, so it made sense to push everything to Sidekiq where jobs could be processed concurrently. But now we have the ability to do this within the consumer itself, so we can remove the Sidekiq dependency and process jobs directly from RabbitMQ.
+
+## Decision
+We have decided to remove Sidekiq from the RabbitMQ queue consumer code, and instead leverage the concurrency features of the [govuk_message_queue_consumer][] gem.
+
+## Status
+Approved
+
+## Consequences
+One useful feature of Sidekiq is that it automatically handles retries. When a job fails, Sidekiq will catch the error and try again multiple times with exponential backoff. The tradeoff we've made here is that we've now had to write code to do this ourselves in the queue consumers, which involves a bit of extra complexity both with regards to the code in this repo and the setup of RabbitMQ.
```

Comment:
I expect it's worth flagging that we've done the extra complexity because RabbitMQ provides no other retry system than a basic automatically retry straight away without extra queue configuration